src/wiki-table-editor.jsx

```diff
@@ -187,29 +231,61 @@ class TableEditor extends React.Component {
   /**
    * Creates a new row at the bottom of the table.
    */
-  addRow() {
+  addRow(rowData) {
     const rows = cloneDeep(this.props.rows);
 
-    let newRow = {
+    rows.push({
+      ...rowData,
       id: this.getNextId()
-    };
+    });
+
+    this.props.setRows(rows);
 
-    // Fill in each required property with an empty string.
+    // Clear the "add new row" row.
+    let newRow = {id: 1};
     this.props.columns.forEach((column) => {
-      if (!column.property) {
-        // Not all columns represent a per-row property. For example, the
-        // "delete button" column.
-        return;
-      };
+       newRow[column.property] = '';
+    });
+    this.setState({ newRow });
+  }
 
-      newRow[column.property] = '';
+  /**
+   * Change the value of one property on one row.
+   */
+  setCell(rowId, property, value) {
+     const index = findIndex(this.props.rows, { id: rowId });
+     const rows = cloneDeep(this.props.rows);
+
+     rows[index][property] = value;
+
+     this.props.setRows(rows);
+  }
+
+  /**
+   * Change the value of a cell in the "add new row" bar.
+   */
+  setNewCell(rowId, property, value) {
+    this.setState({
+      newRow: {
+        ...this.state.newRow,
+        [property]: value
+      }
     });
+  }
 
-    rows.push(newRow);
-    this.props.setRows(rows);
+  onFocus() {
+    this.setState({
+      isEditing: true
+    });
+  }
+
+  onBlur() {
+    this.setState({
+      isEditing: false
+    });
   }
 }
 
 const WikiTableEditor = DragDropContext(HTML5Backend)(TableEditor);
 
-export default WikiTableEditor;
+module.exports = WikiTableEditor;
```

Comment:
Required (I think?) to make webpack expose this as `window.WikiTableEditor` instead of `window.WikiTableEditor.default`